src/contracts/protocol/libraries/logic/LiquidationLogic.sol

```diff
@@ -53,27 +62,126 @@ library LiquidationLogic {
 
   /**
-   * @dev This constant represents below which health factor value it is possible to liquidate
-   * an amount of debt corresponding to `MAX_LIQUIDATION_CLOSE_FACTOR`.
-   * A value of 0.95e18 results in 0.95
+   * @dev This constant represents a base value threshold.
+   * If the total collateral or debt on a position is below this threshold, the close factor is raised to 100%.
+   * @notice The default value assumes that the basePrice is usd denominated by 8 decimals and needs to be adjusted in a non USD-denominated pool.
    */
-  uint256 public constant CLOSE_FACTOR_HF_THRESHOLD = 0.95e18;
+  uint256 public constant MIN_BASE_MAX_CLOSE_FACTOR_THRESHOLD = 2000e8;
+
+  /**
+   * @dev This constant represents the minimum amount of assets in base currency that need to be leftover after a liquidation, if not clearing a position completely.
+   * This parameter is inferred from MIN_BASE_MAX_CLOSE_FACTOR_THRESHOLD as the logic is dependent.
+   * Assuming a MIN_BASE_MAX_CLOSE_FACTOR_THRESHOLD of `n` a liquidation of `n+1` might result in `n/2` leftover which is assumed to be still economically liquidatable.
+   * This mechanic was introduced to ensure liquidators don't optimize gas by leaving some wei on the liquidation.
+   */
+  uint256 public constant MIN_LEFTOVER_BASE = MIN_BASE_MAX_CLOSE_FACTOR_THRESHOLD / 2;
+
+  /**
+   * @notice Reduces a portion or all of the deficit of a specified reserve by burning:
+   * - the equivalent aToken `amount` for assets with virtual accounting enabled
+   * - the equivalent `amount` of underlying for assets with virtual accounting disabled (e.g. GHO)
+   * The caller of this method MUST always be the Umbrella contract and the Umbrella contract is assumed to never have debt.
+   * @dev Emits the `DeficitCovered() event`.
+   * @dev If the coverage admin covers its entire balance, `ReserveUsedAsCollateralDisabled()` is emitted.
+   * @param reservesData The state of all the reserves
+   * @param userConfig The user configuration mapping that tracks the supplied/borrowed assets
+   * @param params The additional parameters needed to execute the eliminateDeficit function
+   */
+  function executeEliminateDeficit(
+    mapping(address => DataTypes.ReserveData) storage reservesData,
+    DataTypes.UserConfigurationMap storage userConfig,
+    DataTypes.ExecuteEliminateDeficitParams memory params
+  ) external {
+    require(params.amount != 0, Errors.INVALID_AMOUNT);
+
+    DataTypes.ReserveData storage reserve = reservesData[params.asset];
+    uint256 currentDeficit = reserve.deficit;
+
+    require(currentDeficit != 0, Errors.RESERVE_NOT_IN_DEFICIT);
+    require(!userConfig.isBorrowingAny(), Errors.USER_CANNOT_HAVE_DEBT);
+
+    DataTypes.ReserveCache memory reserveCache = reserve.cache();
+    reserve.updateState(reserveCache);
+    bool isActive = reserveCache.reserveConfiguration.getActive();
+    require(isActive, Errors.RESERVE_INACTIVE);
```

Comment:
do we wanna continue even if it's paused?

Reply:
yes, as defined on the docs, removing bad debt - even when paused is the intended behavior as there's no downside for the protocol, but only upside.